src/Bicep.Core/Parser/Parser.cs

```diff
@@ -213,6 +213,9 @@ private SyntaxBase ModuleDeclaration()
 
         private Token NewLine()
         {
+            if (Check(TokenType.Comma)) {
+                Expect(TokenType.NewLine, b => b.CommaSeparator());
+            }
```

Comment:
I think we may need something more specifically tailored to being inside an object or an array syntax. Otherwise, the following will most likely give a misleading message:

```
// user is part-way through typing a function call
var myFunc = someFunc(a,
```

Reply:
Good catch

Reply:
And if you end up needing catchall case, just fall back to the expected token behavior that we have today.